cip/CIP2015-05-13-EXISTS.adoc

```diff
@@ -0,0 +1,167 @@
+= CIP2015-05-13 - EXISTENTIAL SUBQUERIES
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Authors:* Andrés Taylor <andres@neotechnology.com>
+
+
+[abstract]
+.Abstract
+--
+This CIP introduces `EXISTS`, a function and a keyword for checking the existence of properties, simple patterns and full subqueries.
+--
+
+toc::[]
+
+== Background
+
+In a way, Cypher already has existential subqueries - pattern predicates allow checking for the existence of subgraphs.
+This form is helpful, but not exhaustive in the types of subqueries that users want to be able to express.
+
+== Proposal
+
+To make this feature more powerful, add a new function named `exists`, and a keyword named the same, which allows for two different predicates:
+
+ * Property existence checking
+ * Subquery existence checking
+
+=== Syntax
+
+----
+expression = <current definition of expression>
+           | property exists
+           | subquery exists
+           | simple subquery exists
+            ;
+
+property exists = "exists", "(", expression ")"
+                ;
+
+subquery exists = "EXISTS", "{", read only clause, { read only clause }, "}" ;
```

Comment:
`readOnlyClause` is not defined. At the very least, it should have a prose definition (e.g. in a comment). At the best, an exhaustive list of clauses.